src/main/java/com/synopsys/integration/blackduck/nexus3/task/inspector/InspectorScanner.java

```diff
@@ -126,133 +150,83 @@ private boolean processAsset(ComponentService componentService, AssetWrapper ass
-
-            ExternalId projectRoot = simpleBdioFactory.createNameVersionExternalId(nexusForge, repositoryName, INSPECTOR_VERSION_NAME);
-            SimpleBdioDocument simpleBdioDocument = simpleBdioFactory.createSimpleBdioDocument(codeLocationName, repositoryName, INSPECTOR_VERSION_NAME, projectRoot, mutableDependencyGraph);
-
-            CodeLocationCreationData<UploadBatchOutput> uploadData = sendInspectorData(inspectorConfiguration.getBdioUploadService(), simpleBdioDocument, simpleBdioFactory, projectNameVersion, codeLocationName);
-
-            Set<String> successfulCodeLocationNames = uploadData.getOutput().getSuccessfulCodeLocationNames();
-            CodeLocationWaitResult.Status status = CodeLocationWaitResult.Status.PARTIAL;
-            if (successfulCodeLocationNames.contains(codeLocationName)) {
-                CodeLocationWaitResult codeLocationWaitResult = inspectorConfiguration.getCodeLocationCreationService().waitForCodeLocations(uploadData.getNotificationTaskRange(), projectNameVersion, successfulCodeLocationNames,
-                    successfulCodeLocationNames.size(), timeOutInSeconds * 5L);
-                status = codeLocationWaitResult.getStatus();
+            return projectBomService.addComponentToProjectVersion(externalId, projectVersionView);
+        } catch (BlackDuckApiException e) {
+            IntegrationRestException integrationRestException = e.getOriginalIntegrationRestException();
+            if (RestConstants.PRECON_FAILED_412 == integrationRestException.getHttpStatusCode()) {
+                // component is already part of the BOM, there is no quick way to determine this ahead of time short of retrieving the whole BOM and searching through the components
+                return Optional.ofNullable(getComponentVersionUrl(componentService, externalId));
             }
-            if (CodeLocationWaitResult.Status.COMPLETE == status) {
-                inspectorMetaDataProcessor.updateRepositoryMetaData(inspectorConfiguration.getBlackDuckService(), blackDuckUrl, projectVersionView, assetWrapperMap, TaskStatus.SUCCESS);
+            throw e;
+        }
+    }
+
+    private String getComponentVersionUrl(ComponentService componentService, ExternalId externalId) throws IntegrationException {
+        Optional<ComponentsView> componentSearchResultView = componentService.getFirstOrEmptyResult(externalId);
+        String componentVersionUrl = null;
+        if (componentSearchResultView.isPresent()) {
+            if (StringUtils.isNotBlank(componentSearchResultView.get().getVariant())) {
+                componentVersionUrl = componentSearchResultView.get().getVariant();
             } else {
-                inspectorMetaDataProcessor.updateRepositoryMetaData(inspectorConfiguration.getBlackDuckService(), blackDuckUrl, projectVersionView, assetWrapperMap, TaskStatus.FAILURE);
+                componentVersionUrl = componentSearchResultView.get().getVersion();
```

Comment:
Should we call get just once and set it to a variable instead of calling it 3 times?